apm-agent-plugins/apm-scala-concurrent-plugin/src/test/scala/co/elastic/apm/agent/scala/concurrent/FutureInstrumentationSpec.scala

```diff
@@ -0,0 +1,91 @@
+package co.elastic.apm.agent.scala.concurrent
+
+import java.util.concurrent.Executors
+
+import co.elastic.apm.agent.MockReporter
+import co.elastic.apm.agent.bci.ElasticApmAgent
+import co.elastic.apm.agent.configuration.{CoreConfiguration, SpyConfiguration}
+import co.elastic.apm.agent.impl.transaction.Transaction
+import co.elastic.apm.agent.impl.{ElasticApmTracer, ElasticApmTracerBuilder}
+import net.bytebuddy.agent.ByteBuddyAgent
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.stagemonitor.configuration.ConfigurationRegistry
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutor, Future}
+
+class FutureInstrumentationSpec extends AnyFlatSpec with Matchers with BeforeAndAfterEach {
+
+  implicit def executionContext: ExecutionContextExecutor =
+    ExecutionContext.fromExecutor(Executors.newFixedThreadPool(1))
+
+  private var reporter: MockReporter = _
+  private var tracer: ElasticApmTracer = _
+  private var coreConfiguration: CoreConfiguration = _
+  private var transaction: Transaction = _
+
+  override def beforeEach: Unit = {
+    reporter = new MockReporter
+    val config: ConfigurationRegistry = SpyConfiguration.createSpyConfig
+    coreConfiguration = config.getConfig(classOf[CoreConfiguration])
+    tracer = new ElasticApmTracerBuilder().configurationRegistry(config).reporter(reporter).build
+    ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install)
+    transaction = tracer.startRootTransaction(null).withName("Transaction").activate()
```

Comment:
I'm unsure on how to start the initial transaction for every test. Since the FutureInstrumentation doesn't create any transactions, I expect that I manually need to activate it here.
Although, I cannot find a similar thing in the `AsyncTraceMethodInstrumentationTest` of the java-concurrent plugin.